### PR TITLE
[Server] Delete the ServerConfig pointers from settings_

### PR DESCRIPTION
**Pull request by gbudau**
Reverted this change.

Co-authored-by: Victor Dedios <victor.dedios@outlook.com>

### CODE REVIEW

**Review by Damrod — APPROVED** [review]
I didn't notice we lost this along the way. Thanks for the fix to both!

File: srcs/server/Server.cpp [hunk 1]
```diff
@@ -20,6 +20,10 @@ Server::~Server() {
 		delete h_it->second;
 	}
 	SyscallWrap::closeWr(listen_sd_ DEBUG_INFO);
+	Server::serverSettingsMap::iterator s_it;
```
Comment (Damrod) [inline srcs/server/Server.cpp:23]: Don't you have to initialize s_it to something like  .begin() ?

Comment (gbudau) [reply srcs/server/Server.cpp]: Good catch! Thank you!

**Review by vdedios — APPROVED** [review]
Good catch @gbudau, I don't know what I was thinking when deleted it :laughing: Thanks!!

**Comment by gbudau** [comment]
Thanks! :raised_hands:

**Pull request merged**